### PR TITLE
Improve integration  test case coverage

### PR DESCRIPTION
**Pull request by panigs7**
# Description
Added test cases to improve integration test coverage.

Improved the coverage % from :
filesystem.go : 67.5% -> 91.9 %
host.go: 90.2% -> 94.3%
metrics.go: 56.1% ->90.2 %
snapshot.go: 65.9% ->94.6%
volume.go: 78% -> 90.8%

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/243 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested using `make go-coverage`
- [x] Verified the coverage `go tool cover -html=gounity_coverprofile.out`

The zip files contain coverage data and logs of integration tests run:
[output.zip](https://github.com/dell/gounity/files/8518208/output.zip)

### CODE REVIEW

File: filesystem_test.go [hunk 1]
```diff
@@ -174,6 +209,23 @@ func createNfsShareTest(t *testing.T) {
...
+
+	snapshotIDTemp = "dummy_snap_1"
+	_, err = testConf.fileAPI.CreateNFSShareFromSnapshot(ctx, nfsShareName+"_by_snap", NFSShareLocalPath, snapshotIDTemp, NoneDefaultAccess)
+	if err == nil {
+		t.Fatalf("Create NFS Share from snapshot with invalid snapshot Id case failed: %v", err)
+	}
+
+	_, err = testConf.fileAPI.CreateNFSShareFromSnapshot(ctx, nfsShareName+"_by_snap", NFSShareLocalPath, snapshotID, NoneDefaultAccess)
+	if err == nil {
+		t.Fatalf("Create NFS Share from snapshot with an existisng nfs share name case failed: %v", err)
```
Comment (rensyct) [inline filesystem_test.go:226]: spell check existisng -> existing

File: test.properties_template [hunk 2]
```diff
@@ -12,4 +12,6 @@ UNITY_POOLID=
 UNITY_NAS_SERVER=
 GOUNITY_DEBUG=true
 HOST_IO_LIMIT_NAME=
+HOST_LIST_NAME= #Comma separated host names to be exported to volumes. Ex: test_host_1,test_host_2 
+TENANT_ID= #Single Tenant ID to be mapped to a host. Ex: tenant_1
```
Comment (rajendraindukuri) [inline test.properties_template:16]: @panigs7 Add a new line at the end of file

**Review by rajendraindukuri — APPROVED** [review]

File: filesystem_test.go (hunk 1 above)
Comment (panigs7) [reply filesystem_test.go]: Done

File: test.properties_template (hunk 2 above)
Comment (panigs7) [reply test.properties_template:16]: Done

**Review by rensyct — APPROVED** [review]

**Review by rajendraindukuri — APPROVED** [review]

**Pull request merged**